### DISCUSSION TITLE
how do we use fastify middleware?

### DISCUSSION THREAD

**Discussion by tsdexter** [gatsby-uc/plugins#226] [Q&A] [answered]
how do I use fastify middleware such as https://github.com/fastify/fastify-multipart/

**Comment by moonmeister** [comment, marked as answer]
There are undocumented plugins within this package that you could use in a custom server that implement multipart. If you have specific examples of what you're doing and if you think it should be implemented in the server we'd probably take a PR.

**Reply by moonmeister** [reply]
Basically you'd be re-implementing  https://github.com/gatsby-uc/plugins/blob/main/packages/gatsby-plugin-fastify/src/plugins/gatsby.ts or https://github.com/gatsby-uc/plugins/blob/main/packages/gatsby-plugin-fastify/src/serve.ts to add this plugin.

**Reply by tsdexter** [reply]
My intention was to use it within a serverless function to submit gravity forms back to wordpress (including file fields) but in the interest of time I will just be posting the `formData` back to the WP instance for now.

**Reply by moonmeister** [reply]
Yeah, that's why I have done previously.

**Reply by tsdexter** [reply]
Yeah, it certainly makes it a bit easier. Using `html-react-parser` allows me to replace the `gform_wrapper` from the post content with a custom `<GravityForm/>` component with a submit handler. This way I don't have to mess around with all the individual fields, styling, etc... just use the HTML gravity outputs. 

I did make the serverless function with authenticated rest api submission though which is working on standard gatsby cloud functions (with minor changes for Request and Response as documented) it should also work in fastify save for the missing multipart handler but there are some other issues with that method such as multi file fields which cannot be submitted via the current API.
